### ISSUE TITLE
設定辞書 `dict[str, Any]` に TypedDict を導入する

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#270]
### Summary

30以上のファイルで `config: dict[str, Any]` が使われ, 設定値へのアクセスに型安全性がない.

### Background

Python ベースの設定ファイル (`configs/rtdetr_coco.py` 等) から読み込んだ設定がすべて `dict[str, Any]` で渡されるため, キーの typo やデータ型の不整合が実行時まで検出できない.

### Current Implementation

```python
# 30+ファイルで使用
def some_function(config: dict[str, Any]) -> ...:
    lr = config["learning_rate"]  # 型不明
```

### Proposed Changes

- 学習用 `TrainingConfig`, 推論用 `InferenceConfig` 等の `TypedDict` を定義する
- 段階的に `dict[str, Any]` を置き換える
- `ConfigLoader` で TypedDict への変換を行う

### Acceptance Criteria

- 主要な設定アクセスパターンが TypedDict で型付けされている
- `mypy` が通る
- 既存テストが通る

### Labels

`refactor`

### Branch

`refactor/config-typeddict`
